### PR TITLE
[terra-section-header] Accordion header jaws fix

### PR DESCRIPTION
**Pull request by sugan2416**
### Summary

Fix for `terra-section-header` JAWS response.

Closes #

### Deployment Link

https://terra-core-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by sugan2416** [comment]
**JAWS accordion-section-header focus :**

- Upon tab navigation to component
tab

- Enter/Space key action on component
action

File: packages/terra-section-header/src/SectionHeader.jsx [hunk 1]
```diff
@@ -140,11 +140,13 @@ class SectionHeader extends React.Component {
     /* eslint-disable jsx-a11y/no-static-element-interactions */
     return (
       <Element {...attributes} onClick={onClick} className={sectionHeaderClassNames} tabIndex="0">
-        <Arrange
-          fitStart={onClick && accordionIcon}
-          fill={<span aria-hidden={(onClick !== undefined)} className={cx('title')}>{headerText}</span>}
-          className={cx('title-arrange')}
-        />
+        <div role="button" aria-expanded={isOpen} tabIndex="-1" aria-label={headerText} className={cx('arrange-wrapper')}>
```
Comment (supreethmr) [inline packages/terra-section-header/src/SectionHeader.jsx:143]: Do we require `tabIndex="-1"` for this div

Comment (sugan2416) [reply packages/terra-section-header/src/SectionHeader.jsx:143]: Yes, without which, jaws doesn't announce state of button

File: packages/terra-core-docs/CHANGELOG.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 * Added
   * Added Accordion Examples for `terra-section-header`.
-  * Updated `terra-section-header` tests and examples.
+  * Updated `terra-section-header` tests and examples
```
Comment (supreethmr) [inline packages/terra-core-docs/CHANGELOG.md:10]:
```suggestion
  * Updated `terra-section-header` tests and examples.
```

Comment (sugan2416) [reply packages/terra-core-docs/CHANGELOG.md]: Changed in https://github.com/cerner/terra-core/pull/3680/commits/33b97e65c97a86d2607cc471fe65233c7cc82a22

File: packages/terra-section-header/src/SectionHeader.module.scss [hunk 3]
```diff
@@ -21,6 +21,7 @@
     padding-bottom: var(--terra-section-header-padding-bottom, 0.5rem);
     padding-top: var(--terra-section-header-padding-top, 0.5rem);
     width: 100%;
+    margin: 0;
```
Comment (supreethmr) [inline packages/terra-section-header/src/SectionHeader.module.scss:24]: why do we need change in margin for section-header ?? I guess we were trying to style only content of section-header

Comment (sugan2416) [reply packages/terra-section-header/src/SectionHeader.module.scss:24]: This was added to suppress the default margin added by heading element (which was causing wdio failures)

**Review by supreethmr — APPROVED** [review]

**Review by saket2403 — APPROVED** [review]

**Pull request merged**